### PR TITLE
[NEW] Create releases tab in the marketplace app info page

### PR DESCRIPTION
**Pull request by rique223**
## Proposed changes (including videos or screenshots)

Added a Releases tab to the app info page of installed marketplace apps. This tab will show all the released versions of a given app with its version number, release date in humanized form, and the changelog of this given release with the information provided by the publisher, this changelog accepts and renders markdown. Also refactored some component names and logic for maintainability reasons.
Demo gif:
app-releases-tab-final

### CODE REVIEW

**Review by graywolf336 — CHANGES_REQUESTED** [review]
Can we please have it only load the release tab and not cause the entire page to seem like it is loading? I expect to have it only load the tab information whenever I click on a tab when that app information is already there.

**Comment by rique223** [comment]
> Can we please have it only load the release tab and not cause the entire page to seem like it is loading? I expect to have it only load the tab information whenever I click on a tab when that app information is already there.

This was a little bug because of a misuse of the router functions, should be solved after this last commit.

File: apps/meteor/app/apps/server/communication/rest.js [hunk 1]
```diff
@@ -528,6 +528,39 @@ export class AppsRestApi {
...
+					if (token) {
+						headers.Authorization = `Bearer ${token}`;
+					}
+
+					let result;
+					try {
+						result = HTTP.get(`${baseUrl}/v1/apps/${this.urlParams.id}`, {
+							headers,
+						});
+						console.log('Rest js versions', result);
```
[the remark this answers is not in the document]
Comment (rique223) [reply apps/meteor/app/apps/server/communication/rest.js:549]: Solved

**Comment by rique223** [comment]
> Awesome, can you update the demo gif?

Sure can do

File: packages/rest-typings/src/apps/index.ts [hunk 2]
```diff
@@ -21,6 +21,13 @@ export type AppsEndpoints = {
 		};
 	};
 
+	'/apps/:id/versions': {
+		GET: () => {
+			apps: App[];
+			success: boolean;
```
Comment (ggazzo) [inline packages/rest-typings/src/apps/index.ts:27]:
```suggestion
```

**Review by ggazzo — DISMISSED** [review]

**Review by graywolf336 — CHANGES_REQUESTED** [review]
When no update information has been provided, let's provide a generic text saying "no release information was provided". You can get with the designers to see what they think would be better.

**Review by graywolf336 — APPROVED** [review]
Looks good!

**Review by graywolf336 — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**